### ISSUE TITLE
[$1000] DEV: Task completed :white_check_mark: is updated only after viewing the task page in second user

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#21761]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
precondition :  Tested on Dev v1.3.32-5
1.  On user A,  go to '+" -> new group
2. pick members including user B and others -> next
3.  go to just created group page
4. go to '+' inside group page -> assign task -> add name
5. leave the assignee and share to default.
6. Now on task page, mark as complete.
7. On user B, go to created group page ( not task)
8. Notice that there's a task created chat with tickmark/check box and a reply chat
9. Notice that it's a task completed by user A, but the tick mark isn't shown. And tick mark appears only when user B, enters and exits task page ( by clicking on replied chat)

## Expected Result:
If task is marked as completed from user A, if should be shown as completed with a :white_check_mark:sign. Please [review this comment](https://github.com/Expensify/App/issues/21761#issuecomment-1627331207) to see how we'd like to implement this. 

## Actual Result: 
Task completed :white_check_mark: is updated only after viewing the task page in second user

## Workaround:
Can the user still use Expensify without this being fixed? Have you informed them of the workaround?

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [x] MacOS / Chrome / Safari
- [x] MacOS / Desktop

**Version Number:** 
**Reproducible in staging?:** DEV v1.3.32-5
**Reproducible in production?:** DEV  v1.3.32-5
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/43996225/ead40a48-d1f9-482c-9cb5-7dcc69620d50

**Expensify/Expensify Issue URL:**
**Issue reported by:** @ashimsharma10
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1687779528925399

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01d584dcc018193cdb</li>
        <li>Upwork Job ID: 1674526730715918336</li>
        <li>Last Price Increase: 2023-07-20</li>
    </ul>
</details>

**Comment by jeet-dhandha** [comment]
I would suggest we hold this issue or close it as we are redesigning task details page.
https://github.com/Expensify/App/issues/19330

**Comment by joekaufmanexpensify** [comment]
Got it! I actually think this might be a dupe of [this issue](https://github.com/Expensify/App/issues/20549).

**Comment by joekaufmanexpensify** [comment]
Once we decide where we are handling this, I will look into whether it makes sense to hold this on the above issue.

**Comment by joekaufmanexpensify** [comment]
Okay, I can reproduce this, so adding external label to get C+ opinion on whether this should be held on [the above issue](https://github.com/Expensify/App/issues/21761#issuecomment-1610769923)/closed..

**Comment by ygshbht** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
In the scenario where User A creates a group and assigns a task, upon marking the task as complete, User B, who is also a member of the group, does not immediately see the task as completed in the group page. The task is displayed as completed for User B only after they navigate to the task page and then return.

### What is the root cause of that problem?
The TaskPreview component determines the isTaskCompleted status based on whether props.taskReport is present or not. If it is present, the status is determined using props.taskReport, otherwise it uses data from props.action. Here's the logic currently used to decide isTaskCompleted:

```
const isTaskCompleted = props.taskReport
    ? props.taskReport.stateNum === CONST.REPORT.STATE_NUM.SUBMITTED && props.taskReport.statusNum === CONST.REPORT.STATUS.APPROVED
    : props.action.childStateNum === CONST.REPORT.STATE_NUM.SUBMITTED && props.action.childStatusNum === CONST.REPORT.STATUS.APPROVED;
```

However, when User B has not yet viewed the task, the props.taskReport object is present but does not contain any data. This leads the isTaskCompleted check to use this empty object instead of the props.action data, causing an incorrect determination of the task's completion status.


### What changes do you think we should make in order to solve the problem?
The most straightforward way to solve this would be to adjust the condition used to determine the isTaskCompleted status. Instead of only checking for the presence of props.taskReport, we can ensure that the object is not empty and contains the required data before using it. This way, we could prevent an empty props.taskReport from causing an incorrect isTaskCompleted determination.

However, if props.taskReport is expected to be undefined when no task report is present, this might indicate a need to address the issue at the data source.

**Comment by joekaufmanexpensify** [comment]
@abdulrahuman5196 When you have a chance, curious if this should be held on [this issue](https://github.com/Expensify/App/issues/21761#issuecomment-1610769923)?

**Comment by abdulrahuman5196** [comment]
>I would suggest we hold this issue or close it as we are redesigning task details page.

@joekaufmanexpensify I don't think we need hold this on the detail view redesign, since this seems to be data propagation issue which might be because of pusher or onyx(Need to understand that).

**Comment by abdulrahuman5196** [comment]
@ygshbht From the proposal here https://github.com/Expensify/App/issues/21761#issuecomment-1614126963, the root cause is correct. 

We can expect task report information to be not present and its intentional as per this comment.

https://github.com/Expensify/App/blob/c0ece67e0edcb725d8f72cf57cd40d98854c552c/src/components/ReportActionItem/TaskPreview.js#L54-L57


Kindly provide more information on what changes you are suggesting to make? It is not clear on what you are trying to propose as changes to fix this.

**Comment by ygshbht** [comment]
@abdulrahuman5196  Thanks for your question. Here's the simplified solution:

The core of the proposed change is to adjust the criteria we use to decide whether to reference **props.taskReport** or props.action in checking if a task is complete. Instead of just verifying if **props.taskReport** exists, we need to confirm that it's not just an empty object. If it is indeed empty, we use **props.action** instead.

Here's the adjusted code:
```
const isTaskReportEmpty = obj => Object.keys(obj).length === 0;

const isTaskCompleted = (props.taskReport && !isTaskReportEmpty(props.taskReport))
    ? props.taskReport.stateNum === CONST.REPORT.STATE_NUM.SUBMITTED && props.taskReport.statusNum === CONST.REPORT.STATUS.APPROVED
    : props.action.childStateNum === CONST.REPORT.STATE_NUM.SUBMITTED && props.action.childStatusNum === CONST.REPORT.STATUS.APPROVED;

```
This change ensures the application accurately determines the completion status of the task, even if **props.taskReport** is an empty object, by relying on **props.action** as a fallback.

**Comment by ygshbht** [comment]
@abdulrahuman5196  
Your question prompted me to crosscheck the solution and, upon further investigation, I've gathered more insights about this issue.

When User A marks a task as completed, User B does not see the update in the Group Chat report screen. However, User B does notice the update when they are in the Report screen of the Task. The reason for this behavior lies in the fact that the Task page listens to a prop named reportActions, while the Group Chat does not for that specific task.

Addressing this discrepancy involves passing reportActions as a prop to the TaskPreview. Here's how the adjusted code looks like with reportActions included as a prop in withOnyx:

```
export default compose(
    withLocalize,
    withOnyx({
        taskReport: {
            key: ({taskReportID}) => `${ONYXKEYS.COLLECTION.REPORT}${taskReportID}`,
        },
        reportActions: {
            key: ({taskReportID}) => `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${taskReportID}`,
            canEvict: false,
            selector: ReportActionsUtils.getSortedReportActionsForDisplay,
        },
    }),
)(TaskPreview);
```

For context, this is the original code that doesn't include reportActions:
```
export default compose(
    withLocalize,
    withOnyx({
        taskReport: {
            key: ({taskReportID}) => `${ONYXKEYS.COLLECTION.REPORT}${taskReportID}`,
        },
    
    }),
)(TaskPreview);

By integrating reportActions into the props, the Group Chat will effectively reflect the updated task status. I hope this clears up the confusion, and thank you for bringing this to my attention. It's these kinds of discussions that bring continuous improvement in our work. Do let me know if you have any further queries.

```

**Comment by abdulrahuman5196** [comment]
@ygshbht I don't think this is required https://github.com/Expensify/App/issues/21761#issuecomment-1617131189

Anyways the suggestion i would do on this proposal is

1) The task report is coming as {} from here which is expected if task is not viewed https://github.com/Expensify/App/blob/c0ece67e0edcb725d8f72cf57cd40d98854c552c/src/components/ReportActionItem/TaskPreview.js#L49
2) Instead of `props.taskReport`, we can use `props.taskReport.stateNum and props.taskReport.statusNum`(Any other different style of check is also fine) https://github.com/Expensify/App/blob/c0ece67e0edcb725d8f72cf57cd40d98854c552c/src/components/ReportActionItem/TaskPreview.js#L57
3) Or we can mark the default props as null here and add proper lodashGet throughout the file(you could find references other places of the app). https://github.com/Expensify/App/blob/8b7a65dcd36fa2b47ffb39c69f2cb294b94a7171/src/components/ReportActionItem/TaskPreview.js#L49

So the solution change would be either 2) or 3)

**Comment by ygshbht** [comment]
@abdulrahuman5196 You might be correct, and I appreciate your critical input. Upon reviewing my previous proposal, I've realized it may not entirely solve the issue at hand due to task report actions being undefined.

As for your suggestions:

2. Could you please elaborate on how checking for **props.taskReport.stateNum** and **props.taskReport.statusNum** would help to address this issue? Since **props.taskReport** can be an empty object, these properties would be **undefined**.

3. For your third suggestion, you mentioned marking the default props as **null** and using **lodashGet** throughout the file. I would appreciate if you could provide a bit more detail on how this approach would effectively handle the scenario where props.taskReport is an empty object. I'd like to add that i see **props.action** too not containing all the necessary data most of the times.

Your suggestions made me rethink the approach. In scenarios where **props.taskReport** might be empty, we could consider fetching the report in a similar manner as it's done in the ReportScreen (GroupChat page). This way, we ensure that the required details are populated correctly.

For reference, the function being used in ReportScreen is as follows:
```
fetchReportIfNeeded() {
    const reportIDFromPath = getReportID(this.props.route);

    if (!reportIDFromPath) {
        return;
    }

    if (this.props.report.reportID && this.props.report.reportID === getReportID(this.props.route)) {
        return;
    }

    Report.openReport(reportIDFromPath);
}
```
I value your insight and look forward to your response!

**Comment by abdulrahuman5196** [comment]
@ygshbht you don't need to fetch the report again. 
If `props.taskReport` is empty, the same information will be available in `props.action` which is a report action. For example, if information is not present in `props.taskReport.stateNum` the same information will be present in `props.action.childStateNum`

That's why for task title it is kept as as `const taskTitle = props.taskReport.reportName || props.action.childReportName;`

**Comment by ygshbht** [comment]
@abdulrahuman5196 
I appreciate your prompt response and further clarification.

I completely agree that if **props.taskReport** is empty, ideally we should be able to retrieve the same information from **props.action**. However, during my analysis, it seems that **props.action** does not include **childStateNum** and **childStatusNum** initially, which are the properties used to determine the task completion status.

I understand your point about not needing to fetch the report again. However, given this, do you have any other suggestions?

![image](https://github.com/Expensify/App/assets/65986357/7aa358be-25c4-4409-8e90-df41882f1db2)

**Comment by abdulrahuman5196** [comment]
>it seems that props.action does not include childStateNum and childStatusNum initially, which are the properties used to determine the task completion status.

Could you complete/cancel the task and see this information is populated, i think so?

**Comment by abdulrahuman5196** [comment]
@thienlnam Curious question on this, why do we even check and use `props.taskReport` which is not guaranteed to have task information whereas the `props.action`(report action of parent) is guaranteed to have information to display the task preview? Any specific case that is solving or we can just use the `props.action` data?

https://github.com/Expensify/App/blob/c0ece67e0edcb725d8f72cf57cd40d98854c552c/src/components/ReportActionItem/TaskPreview.js#L57-L63

**Comment by thienlnam** [comment]
It was because props.action wasn't always up to date. The task report would always have the latest data when an action was taken. Like when you edit a task title we pushed the update to clients subscribed to the task but did not update the parent reportAction. However, we've updated a lot of the flows so we may be able to just rely on the parentReportAction now. But if now, let me know and we can get an issue created to update all flows to update the parentReportAction as well as the taskReport

**Comment by ygshbht** [comment]
> > it seems that props.action does not include childStateNum and childStatusNum initially, which are the properties used to determine the task completion status.
> 
> Could you complete/cancel the task and see this information is populated, i think so?

@abdulrahuman5196  Unfortunately, I tried both scenarios - completing and canceling the task - and in neither case was the information populated.

**Comment by ygshbht** [comment]
> It was because props.action wasn't always up to date. The task report would always have the latest data when an action was taken. Like when you edit a task title we pushed the update to clients subscribed to the task but did not update the parent reportAction. However, we've updated a lot of the flows so we may be able to just rely on the parentReportAction now. But if now, let me know and we can get an issue created to update all flows to update the parentReportAction as well as the taskReport

You can definitely opt to keep props.action updated to latest events. If this course is taken and the updates are implemented, I can modify the isTaskCompleted check to correctly utilize data from props.action.

Alternatively, if you don't opt for this approach, we could also fetch the report in a similar manner as is done on the GroupChat page/ ReportScreen component (comment https://github.com/Expensify/App/issues/21761#issuecomment-1618780089).

**Comment by joekaufmanexpensify** [comment]
Still discussing next steps for a proposal here.

**Comment by joekaufmanexpensify** [comment]
@abdulrahuman5196 Curious if you think we're close to being able to accept the existing proposal here?

**Comment by joekaufmanexpensify** [comment]
Pending confirmation for next steps on existing proposal.

**Comment by abdulrahuman5196** [comment]
Have couple of doubts. Will check again and raise tomorrow morning.

**Comment by abdulrahuman5196** [comment]
>Unfortunately, I tried both scenarios - completing and canceling the task - and in neither case was the information populated.

@ygshbht 
<img width="1512" alt="Screenshot 2023-07-08 at 7 42 17 PM" src="https://github.com/Expensify/App/assets/46707890/b056100a-9d2d-40c7-814c-4886574e1ce6">

**Comment by abdulrahuman5196** [comment]
> It was because props.action wasn't always up to date. The task report would always have the latest data when an action was taken. Like when you edit a task title we pushed the update to clients subscribed to the task but did not update the parent reportAction. However, we've updated a lot of the flows so we may be able to just rely on the parentReportAction now. But if now, let me know and we can get an issue created to update all flows to update the parentReportAction as well as the taskReport

@thienlnam Got it. I am also currently getting the task status properly in the report action itself. But have to check if its coming properly in other flows as well.

But I would suggest to use the report action of the parent in this case instead of fetching the task report itself. It will become intensive action. Like if i have 100 tasks in a parent chat report, it will fetch 100 tasks reports to show the preview in the chat report which is unwanted IMO.

I think as part of this issue, we can have contributors proposal to target,
1) Migrate TaskPreview to use parent report action instead of task report data.
2) Find if any other places if the parent report is not populated, if so we can create a issue to update it in backend.


Kindly let me know your thoughts on this.

**Comment by ygshbht** [comment]
> > Unfortunately, I tried both scenarios - completing and canceling the task - and in neither case was the information populated.
> 
> @ygshbht <img alt="Screenshot 2023-07-08 at 7 42 17 PM" width="1512" src="https://user-images.githubusercontent.com/46707890/251984765-b056100a-9d2d-40c7-814c-4886574e1ce6.png">

I can definitely migrate it to use parent action instead of task report data.

However, as i see currently, childStateNum and childStatusNum are only updated on page refresh or maybe when you move to that report from some other screen. They don't seem to update in real time 
![image](https://github.com/Expensify/App/assets/65986357/f3e1cbbb-a344-4599-a2e4-4e4a5a31cd47)

**Comment by joekaufmanexpensify** [comment]
Still discussing proposal

**Comment by ygshbht** [comment]
> But I would suggest to use the report action of the parent in this case instead of fetching the task report itself. It will become intensive action. Like if i have 100 tasks in a parent chat report, it will fetch 100 tasks reports to show the preview in the chat report which is unwanted IMO.

@abdulrahuman5196 That's a really insightful consideration. The good news here, though, is that the task of fetching needs to be done only once. After that, the system will continuously listen for events. Once a task report has been fetched initially, the system wouldn't need to re-fetch it, since the information will be retained and available for future use.

```
/** You can remove this as per your preference if report.action starts getting updated in real time */
    React.useEffect(() => {
        if (!_.isEmpty(props.taskReport)){
            Report.openReport(reportId);
        }
    }, [props.taskReport]);
```
In the event that there's an upgrade in the future and report.action starts getting updated in real time, if you prefer to not fetch the report, the transition would be smooth and efficient. You would simply have to remove the block of code above.

Please see the video below of task updating in real time once you view it (once it is fetched): 

https://github.com/Expensify/App/assets/65986357/194c1d40-48ff-42bf-b835-7f09e599d9ee

**Comment by joekaufmanexpensify** [comment]
Still discussing proposal

**Comment by joekaufmanexpensify** [comment]
@abdulrahuman5196 Curious for your thoughts on this updated proposal, and if you're thinking we'll ultimately be able to use it?

**Comment by abdulrahuman5196** [comment]
Hi, Sorry for the delay, Will update on the proposal in the morning.

**Comment by abdulrahuman5196** [comment]
@thienlnam Could you kindly confirm on this? https://github.com/Expensify/App/issues/21761#issuecomment-1627331207

**Comment by thienlnam** [comment]
Your comment makes sense, and I agree. Task previews are kind of tricky but have a similar need as the thread count.

Everyone in the parent report will need to see the updated task details, and the only way to verify that happens is by ensuring the reportAction details are up to date. Not everyone will join the task report and therefore will only have the task details from the reportAction

**Comment by abdulrahuman5196** [comment]
@ygshbht Can you update your proposal as per discussion here ? https://github.com/Expensify/App/issues/21761#issuecomment-1627331207

**Comment by abdulrahuman5196** [comment]
@joekaufmanexpensify These 2 implementations are the expectation from this issue - https://github.com/Expensify/App/issues/21761#issuecomment-1627331207

If possible kindly add this comment in the OP, so new contributors are aware of this. And as of now there isn't any proposal for the fix. 

I have requested @ygshbht  to update their proposal. If that doesn't pan out we have to raise the bounty.

**Comment by ygshbht** [comment]
@abdulrahuman5196, Sure. Considering the recent modifications to the condition, where we check _!\_.isEmpty(props.taskReport)_ before using _taskReport_ or _parentAction_, it indeed appears beneficial to eliminate the taskReport reference altogether. Such a change would streamline the code, improve the architecture, and also eliminate any potential confusion, thereby reinforcing the reliability of the status check. Should I proceed with this amendment to the proposal?

**Comment by joekaufmanexpensify** [comment]
@abdulrahuman5196 Sounds good, updated OP to link to that comment!

**Comment by joekaufmanexpensify** [comment]
Not overdue

**Comment by joekaufmanexpensify** [comment]
@abdulrahuman5196 I think we're waiting for your input on [this question](https://github.com/Expensify/App/issues/21761#issuecomment-1634764692) before @ygshbht updates their proposal

**Comment by abdulrahuman5196** [comment]
@ygshbht 

>Sure. Considering the recent modifications to the condition, where we check !_.isEmpty(props.taskReport) before using taskReport or parentAction, it indeed appears beneficial to eliminate the taskReport reference altogether. Such a change would streamline the code, improve the architecture, and also eliminate any potential confusion, thereby reinforcing the reliability of the status check. Should I proceed with this amendment to the proposal?


That is the requirement 1 mentioned here https://github.com/Expensify/App/issues/21761#issuecomment-1627331207. We shouldn't use `taskReport` in `TaskPreview`

>Should I proceed with this amendment to the proposal?

For future cases, kindly don't wait to amend proposal. You can always update the proposal or add alternate solutions with the amendment.

**Comment by ygshbht** [comment]
Thank @abdulrahuman5196 for the clarification. 
## Proposal
[Updated](https://github.com/Expensify/App/issues/21761#issuecomment-1614126963)

**Comment by abdulrahuman5196** [comment]
@ygshbht you have to check on both the requirements mentioned and update the proposal.

**Comment by ygshbht** [comment]
## Proposl

[Updated](https://github.com/Expensify/App/issues/21761#issuecomment-1614126963)

I appreciate the opportunity to provide further insights into this issue. If it's considered part of the task, I've gladly done this additional exploration.

After navigating through the app and combing through the codebase, I did my best to identify any potential problems where the parent report might not be populated. From the aspects I tested - such as replying in threads (which correctly updates the reply count in the action view) and requesting additional funds (which also correctly updates the amount in the action view) - things seem to be functioning as expected.

I've made an effort to scrutinize as many scenarios as possible; however, given the app's considerable scope, there could still be hidden aspects that might have been overlooked. However I did find one thing. **Similar to task completion status, the task title doesn't seem to update either. These two could could be fixed together.**

**Comment by joekaufmanexpensify** [comment]
Pending @abdulrahuman5196  review of @ygshbht's updated proposal!

**Comment by joekaufmanexpensify** [comment]
Proposal still pending review

**Comment by abdulrahuman5196** [comment]
@ygshbht I am still not able to see full picture of how your solution would solve the entire requirements we have mentioned.

**Comment by abdulrahuman5196** [comment]
@joekaufmanexpensify I think we should take next steps here.

**Comment by ygshbht** [comment]
> @ygshbht I am still not able to see full picture of how your solution would solve the entire requirements we have mentioned.

Thanks for the question @abdulrahuman5196. 

The issue originated from both frontend and backend code. Initially, `props.taskReport` [was checked to see if it was defined](https://github.com/Expensify/App/blob/c0ece67e0edcb725d8f72cf57cd40d98854c552c/src/components/ReportActionItem/TaskPreview.js#L56-L61), but it should have been checked to see if it was either an empty or complete `taskReport `object. Post my initial [proposal](https://github.com/Expensify/App/issues/21761#issuecomment-1614126963), the condition was updated to check for an empty `taskReport ` [object](https://github.com/Expensify/App/blob/d3a13d60af2b080b5b8891b37c090e6314c490e0/src/components/ReportActionItem/TaskPreview.js#L64-L68) | https://github.com/Expensify/App/issues/21761#issuecomment-1634764692. Fixing this on the backend now correctly updates task status on the frontend.


My latest proposal, based on your [feedback](https://github.com/Expensify/App/issues/21761#issuecomment-1627331207), does two things:

1. It refactors the frontend code by removing the` props.taskReport` reference, **simplifying it without changing the app's behavior**.
2. It addresses your second point to find similar issues with `props.action`. As per my findings https://github.com/Expensify/App/issues/21761#issuecomment-1643432583, the app seems to work fine in other places. But, similar to the status, **the title for TaskPreview also doesn't seem to update without reloading**.

**In short, my proposal will improve the app's code efficiency and reliability without affecting its current functionality. Plus, I've also identified a similar issue for future consideration.**

That said, it's entirely understandable if the decision is made not to proceed with this proposal or make any further frontend changes, as it wouldn't alter the app's behavior. It wouldn't be an unreasonable choice at all.

**Comment by abdulrahuman5196** [comment]
Will check on the updated proposal in a while

**Comment by joekaufmanexpensify** [comment]
Floated the issue to Callstack as well.

**Comment by rezkiy37** [comment]
Hi, I'm Michael (Mykhailo) from Callstack and I would like to work in this issue.

**Comment by abdulrahuman5196** [comment]
On @Nodebrute 's proposal its a good start - https://github.com/Expensify/App/issues/21761#issuecomment-1646003200

What about the second requirement mentioned here? https://github.com/Expensify/App/issues/21761#issuecomment-1627331207. Because if we want to make backend changes then frontend has to wait on the same.

**Comment by abdulrahuman5196** [comment]
> On @Nodebrute 's proposal its a good start - [#21761 (comment)](https://github.com/Expensify/App/issues/21761#issuecomment-1646003200)
> 
> What about the second requirement mentioned here? [#21761 (comment)](https://github.com/Expensify/App/issues/21761#issuecomment-1627331207). Because if we want to make backend changes then frontend has to wait on the same.

@Nodebrute Ping on the same

**Comment by joekaufmanexpensify** [comment]
Discussing next steps [here. ](https://expensify.slack.com/archives/C02NK2DQWUX/p1690570243601399?thread_ts=1690402036.302879&cid=C02NK2DQWUX)

**Comment by abdulrahuman5196** [comment]
@Nodebrute Gentle ping

**Comment by abdulrahuman5196** [comment]
> > 2. Find if any other places if the parent report is not populated, if so we can create a issue to update it in backend.
> 
> @abdulrahuman5196 I would appreciate it if you could provide more details or further explanation regarding this.

@Nodebrute Basically on your proposal(for anyone's) to work `props.action.childStateNum` and similar properties inside `props.action`(report action) should have proper task information for it to update the taskPreview. But we are not sure if in all task updation cases the report action is updated. We think some cases its not updating properly, we need to know that cases and update backend accordingly.

More reference here - https://github.com/Expensify/App/issues/21761#issuecomment-1618957628

**Comment by joekaufmanexpensify** [comment]
Still discussing in Slack

**Comment by joekaufmanexpensify** [comment]
Confirming if Callstack still wants to take this one.

**Comment by rezkiy37** [comment]
I already have a proposal and it is in our internal review.

**Comment by joekaufmanexpensify** [comment]
Great, thanks!

**Comment by rezkiy37** [comment]
Melvin, I posted a proposal and someone should review it soon.

**Comment by abdulrahuman5196** [comment]
@rezkiy37 

>On CompleteTask and ReopenTask requests (commands), the backend does not include important updated values: childStateNum and childStatusNum. Moreover, these values have been updated by these commands. The backend must respond with all updated values, includes childStateNum and childStatusNum.

I am not able to understand this. 

>On CompleteTask and ReopenTask requests (commands), the Pusher also must include all updated values. Now, it does not include updated childStateNum and childStatusNum.

For cases where backend is not properly providing data, can you provide use cases/steps to repro? So that we can test the same and create backend specific issues on the same?

I think only after backend is fixed, we can update frontend. Or it could cause data loss and regression.

**Comment by rezkiy37** [comment]
@abdulrahuman5196, I will send a clarification message tomorrow 🙂 
Thank you!

**Comment by joekaufmanexpensify** [comment]
@abdulrahuman5196 curious for your thoughts on this?

**Comment by abdulrahuman5196** [comment]
Thank you @rezkiy37. @thienlnam Could you check on this? We should fix these cases in backend before making the UI fix.

Just linking the older comment we discussed - https://github.com/Expensify/App/issues/21761#issuecomment-1627331207

**Comment by thienlnam** [comment]
We already have a BE issue to return more parameters for unread indicator - I believe that same PR https://github.com/Expensify/Expensify/issues/307566 should also return more parameters

**Comment by joekaufmanexpensify** [comment]
On hold

**Comment by joekaufmanexpensify** [comment]
Same

**Comment by joekaufmanexpensify** [comment]
Held

**Comment by joekaufmanexpensify** [comment]
Same

**Comment by joekaufmanexpensify** [comment]
Same

**Comment by thienlnam** [comment]
Held PR should be live! We can try retesting this

**Comment by joekaufmanexpensify** [comment]
@thienlnam I just re-tested this and this behavior still exists. The task page only updates after user 2 leaves and goes back into the chat. Though I'm not super sold we need to do anything to change this. IIRC, there had been a discussion about not really considering these "live updating" issues bugs in Slack. 

https://github.com/Expensify/App/assets/54866469/c0cc0a7b-5503-4995-b0cc-4949f061fdb6

**Comment by thienlnam** [comment]
Yeah, I'm not super inclined to investigate on the BE either - focused on a few other projects right now. Let's just close this?
